### PR TITLE
Document stepper

### PR DESCRIPTION
**Pull request by alshakero**
#### Changes proposed in this Pull Request

This documents the changes in this [PR](https://github.com/Automattic/wp-calypso/pull/61732).

### CODE REVIEW

File: client/landing/stepper/README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
+<img src="https://user-images.githubusercontent.com/17054134/159939643-4a3a7893-ab38-4223-ba5d-5dfe64282f07.png" alt="Stepper logo" width="400">
+
+# Stepper Signup and Onboarding Framework
+The stepper framework is a new framework for quickly spinning up sign up flows. It's meant to make non-linearly easy without sacrificing agility ✨. 
+
+## Non-linearity
+It has been tricky for us to create flows that input-driven steps. Stepper makes it easy by allowing flows to create their own two hooks `useSteps`, and `useStepNavigation`. These hooks have access to the state of of the flow so they can make decisions based on that.
```
Comment (escapemanuele) [inline client/landing/stepper/README.md]: `create flows that` ... `input-driven steps`: missing something
`to the state of of the flow`: double of

File: client/landing/stepper/README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	useStepNavigation( currentStep, navigate ) {
+		const goBack = () => {
+			if ( currentStep === 'domain' ) {
+				navigate( 'design' );
+			} else {
+				navigate( 'domain' );
+			}
+		};
+		const goNext = goBack;
+		return { goNext, goBack, goToStep };
```
Comment (escapemanuele) [inline client/landing/stepper/README.md]: goToStep is not defined in the example

File: client/landing/stepper/README.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+Stepper aims to create a big `steps-repository` that contains the steps and allows them to be recycled and reused. Every step you create is inherently reusable by any future flow. Because steps are like components, they're not parts of the flows, flows just happen to use them.
+
+This creates a couple of restrictions.
+
+## State management
+Steps shouldn't have any props other than `navigation` prop which contains the return value of `useStepNavigation`. This object (`navigation`) allows the step to `submit` when done or to move to other steps or to `goNext`, `goBack`, etc..
+
+The rationale behind this is the following:
+
+1. **Small API**: when steps have such little API surface area. They can be reused easily.
```
Comment (escapemanuele) [inline client/landing/stepper/README.md]: API surface area, they can

File: client/landing/stepper/README.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+2. **Centralized state**: the `useStepNavigation` hook makes decisions based on the entire flow's state, so the state has to be centralized, i.e: lives entirely in the `onboard` store (or any store inside `packages/data-stores`).
+
+All the state should be either the onboarding store, or in the internal state of the step. This creates a flat data tree that is exactly one level deep. This allows the steps to be easy to make and to reuse without any added effort.
+
+It takes five minutes to add a new field to the onboarding store. Such as `intent` or `storeType`. 
+
+## Preferring packages
+When creating new code, it's highly-encouraged to put it packages, whether existing like `onboarding` or new packages that you create. This allows us to create flows in the future that live in wp-admin or wherever.
```
Comment (escapemanuele) [inline client/landing/stepper/README.md]: put it _into_ packages

**Review by escapemanuele — CHANGES_REQUESTED** [review]
Great description of the framework! Only a few changes to make and then we can ship that!

File: client/landing/stepper/README.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
+<img src="https://user-images.githubusercontent.com/17054134/159939643-4a3a7893-ab38-4223-ba5d-5dfe64282f07.png" alt="Stepper logo" width="400">
```
Comment (arcangelini) [inline client/landing/stepper/README.md:1]: Haha I love the logo 👍 👍

File: client/landing/stepper/README.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+# Stepper Signup and Onboarding Framework
+The stepper framework is a new framework for quickly spinning up sign up flows. It's meant to make non-linearly easy without sacrificing agility ✨.
```
Comment (escapemanuele) [inline client/landing/stepper/README.md]: Maybe `non-linearity`?

Comment (arcangelini) [inline client/landing/stepper/README.md]:
```suggestion
The stepper framework is a new framework for quickly spinning up sign-up flows. This is a non-linear solution that doesn't sacrifice agility ✨. 
```

This is a bit wordy and doesn't read well. Just a suggestion.

File: client/landing/stepper/README.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+};
+```
+
+## Reusability
+Stepper aims to create a big `steps-repository` that contains the steps and allows them to be recycled and reused. Every step you create is inherently reusable by any future flow. Because steps are like components, they're not parts of the flows, flows just happen to use them.
+
+This creates a couple of restrictions.
+
+## State management
+Steps shouldn't have any props other than `navigation` prop which contains the return value of `useStepNavigation`. This object (`navigation`) allows the step to `submit` when done or to move to other steps or to `goNext`, `goBack`, etc..
```
Comment (arcangelini) [inline client/landing/stepper/README.md]:
```suggestion
Steps shouldn't have any props other than the `navigation` prop which contains the return value of `useStepNavigation`. This object (`navigation`) allows the step to `submit` when done or to move to other steps or to `goNext`, `goBack`, etc..
```

I think this needs a 'the'

File: client/landing/stepper/README.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+The rationale behind this is the following:
+
+1. **Small API**: when steps have such little API surface area. They can be reused easily.
+2. **Centralized state**: the `useStepNavigation` hook makes decisions based on the entire flow's state, so the state has to be centralized, i.e: lives entirely in the `onboard` store (or any store inside `packages/data-stores`).
+
+All the state should be either the onboarding store, or in the internal state of the step. This creates a flat data tree that is exactly one level deep. This allows the steps to be easy to make and to reuse without any added effort.
```
Comment (arcangelini) [inline client/landing/stepper/README.md]:
```suggestion
All the state should be either in the onboarding store or the internal state of the step. This creates a flat data tree that is exactly one level deep and allows steps to be easily created and re-used.
```

Another wording suggestion.

File: client/landing/stepper/README.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+        return { goNext, goBack };
+	},
+};
+```
+
+## Reusability
+Stepper aims to create a big `steps-repository` that contains the steps and allows them to be recycled and reused. Every step you create is inherently reusable by any future flow. Because steps are like components, they're not parts of the flows, flows just happen to use them.
+
+This creates a couple of restrictions.
+
```
Comment (arcangelini) [inline client/landing/stepper/README.md:81]:
```suggestion

To maintain the reusability and simplicity of this framework it is important that flow-specific styling changes be made to a `flow` stylesheet or in `global.scss`. Each step should have the basic styling necessary to operate on its own just like a package.
```

Maybe best to add something about styling as well?

**Review by arcangelini — APPROVED** [review]
Just some nit-picky comments and suggestions. Great work on this.

**Comment by alshakero** [comment]
Thanks a lot guys! Perfect feedback.
